Library/Homebrew/extend/version-parser.rb

```diff
@@ -0,0 +1,52 @@
+# typed: strict
+# frozen_string_literal: true
+
+# Class handling architecture-specific version information.
+#
+# @api private
+class VersionParser
```

Comment:
```suggestion
class Homebrew::BumpVersionParser
```

or similar. Feels like it should be:
- in Homebrew module
- reference `Bump` in the name

Also, don't put it in `extend` as it's not extending an existing class